core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -998,6 +998,32 @@ public void onFailure(Throwable t)
                     errorScheduledExecutor.schedule(() -> doScheduleAsyncCleanupRequest(cleanupBackoff, request, action), delayNanos, NANOSECONDS);
                 }
             }
+
+            private void fatalAsyncCleanupFailure(Throwable cause)
```

Comment:
This method parameter should be a `TrinoTransportException`, since that's the only valid way to call this code (no "partial failures" are allowed to be considered fatal).